core/controllers/reader.py

```diff
@@ -100,21 +100,45 @@ def test_can_play(self, exploration_id, **kwargs):
     return test_can_play
 
 
-def classify_hard_rule(state, params, input_type, normalized_answer, fs):
+def _evaluate_rule(rule_spec, rule_description, context_params, answer):
+    """Evaluates a rule spec. Returns a float between 0.0 and 1.0."""
+    param_list = []
+    param_defns = rule_domain.get_param_list(rule_description)
+    for (param_name, obj_cls) in param_defns:
+        parsed_param = rule_spec.inputs[param_name]
+        if isinstance(parsed_param, basestring) and '{{' in parsed_param:
+            parsed_param = jinja_utils.parse_string(
+                parsed_param, context_params, autoescape=False)
+        normalized_param = obj_cls.normalize(parsed_param)
+        param_list.append(normalized_param)
+
+    # NOTE TO DEVELOPERS: These lines will not work -- they originally
```

Comment:
This function essentially needs to be removed, along with all of the classify functions below. The only backend dependency on backend classification now is submit_answer, which will be removed as part of the answer migrations work.